### PR TITLE
feat: support netlink mode

### PR DESCRIPTION
**Pull request by fengxsong**
Signed-off-by: fengxsong <fengxsong@outlook.com>

refactor lvscare to enable netlink mode, just like `kube-proxy`, most of the methods are referenced from the `kube-proxy`.

### CODE REVIEW

File: staging/src/github.com/labring/lvscare/pkg/sysctl/sysctl.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
+/*
```
Comment (zzjin) [inline staging/src/github.com/labring/lvscare/pkg/sysctl/sysctl.go:1]: this file is exactly same as k8s 's source.
 so maybe we can just import `"k8s.io/component-helpers/node/util/sysctl"` as module to use, but not copy code straight here?

File: staging/src/github.com/labring/lvscare/pkg/ipset/ipset.go [hunk 2]
```diff
@@ -0,0 +1,528 @@
+/*
```
Comment (zzjin) [inline staging/src/github.com/labring/lvscare/pkg/ipset/ipset.go:1]: Just import `"k8s.io/kubernetes/pkg/util/ipset"` as well

File: controllers/infra/go.mod [hunk 3]
```diff
@@ -88,19 +86,21 @@ require (
...
 	k8s.io/kube-openapi v0.0.0-20220328201542-3ee0da9b0b42 // indirect
 	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9 // indirect
 	sigs.k8s.io/json v0.0.0-20211208200746-9f7c6b3444d2 // indirect
-	sigs.k8s.io/kustomize/kyaml v0.10.15 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-require github.com/labring/sealos v0.0.0-00010101000000-000000000000
+require (
```
Comment (zzjin) [inline controllers/infra/go.mod:101]: Do not change this require as this two lines are just for sealos usage.
(Maybe `go mod tidy` auto changed this line, pls fix it by hand: move `kyaml` to above lines)

Comment (fengxsong) [reply controllers/infra/go.mod]: done

**Comment by cuisongliu** [comment]
You can provide relevant test cases.

File: staging/src/github.com/labring/lvscare/README.md [hunk 4]
```diff
@@ -1,170 +1,136 @@
...
+Run lvscare as a static pod on every kubernetes workers. **now control planes are also supported with `link` mode.**
+
+```bash
+lvscare care --vs 10.103.97.12:6443 --rs 192.168.0.2:6443 --rs 192.168.0.3:6443 --rs 192.168.0.4:6443 --interval 5 --mode link
 ```
 
-- -t every 5s check the real server port
-  - --health-port "6443" if not return 200 OK, remove the realserver
-  - --health-path "/healthz" if not return 200 OK, remove the realserver
+- --mode defaults to `route`, from my test case seems `route` mode doesn't make sense..
```
Comment (fanux) [inline staging/src/github.com/labring/lvscare/README.md:25]: what dose mode toute and mode link means?

**Review by fanux — APPROVED** [review]

File: staging/src/github.com/labring/lvscare/README.md [hunk 5]
```diff
@@ -1,170 +1,136 @@
...
 
-### Test
+Check with `lvscare care --help` command for more options.
 
-If the realserver is on the same host, you need bind vip on a interface
+### Test
 
-```
-ip link add sealyun-ipvs0 type dummy
-ip  addr add 10.103.97.12/32 dev sealyun-ipvs0
```
Comment (fanux) [inline staging/src/github.com/labring/lvscare/README.md:36]: add iptables rules

File: staging/src/github.com/labring/lvscare/care/iptables.go [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+	}
+	for i := range addresses {
+		if _, err := nl.EnsureAddressBind(addresses[i], ifaceName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (impl *iptablesImpl) ensureIpset() error {
```
Comment (fanux) [inline staging/src/github.com/labring/lvscare/care/iptables.go:190]: add ipset into lvscare docker image

File: staging/src/github.com/labring/lvscare/care/runner.go [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+func (r *runner) ValidateAndSetDefaults() error {
+	for _, iter := range []interface{}{r.options, r.prober} {
+		if validator, ok := iter.(flagValidator); ok {
+			if err := validator.ValidateAndSetDefaults(); err != nil {
+				return err
+			}
+		}
+	}
+	r.proxier = NewProxier(r.options.scheduler, r.options.Interval, r.prober, r.periodicRun)
```
Comment (fanux) [inline staging/src/github.com/labring/lvscare/care/runner.go:147]: interval * Senconds

**Review by fanux — APPROVED** [review]

**Comment by fanux** [comment]
https://github.com/labring/lvscare/pull/29/files

File: staging/src/github.com/labring/lvscare/care/proxier.go [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+			go func(rs endpoint) {
+				defer wg.Done()
+				probeErr := p.prober.Probe(rs.IP, strconv.Itoa(int(rs.Port)))
+				rSrv, err := p.getRealServer(vSrv, p.buildRealServer(&rs))
+				if err != nil {
+					logger.Warn("Failed to get real server: %v", err)
+					return
+				}
+				if probeErr != nil {
+					logger.Debug("probe: %v", probeErr)
```
[the remark this answers is not in the document]
Comment (fengxsong) [reply staging/src/github.com/labring/lvscare/care/proxier.go]: 5ed70fa83528553eb4e71e33afd7696de1e7111c

File: staging/src/github.com/labring/lvscare/README.md (hunk 5 above)
Comment (fengxsong) [reply staging/src/github.com/labring/lvscare/README.md:36]: 3294c7bcac5d67c65d27b587ef2b3458a19a1acc

File: staging/src/github.com/labring/lvscare/care/iptables.go (hunk 6 above)
Comment (fengxsong) [reply staging/src/github.com/labring/lvscare/care/iptables.go:190]: 679197db0854f73d2711cddede589ddeebf5cb6e

File: staging/src/github.com/labring/lvscare/care/runner.go (hunk 7 above)
Comment (fengxsong) [reply staging/src/github.com/labring/lvscare/care/runner.go]: d0ebad39927fe900bdefe7fd16cd78461fd1a79d

**Comment by cuisongliu** [comment]
```
./sealos load -i patch-arm64.tar
./sealos images
./sealos pull labring/kubernetes:v1.24.3
./sealos images  
./sealos run labring/kubernetes:v1.24.3 docker.io/labring/patch:868b942-arm64 --single
./sealos add --nodes 192.168.64.25
./sealos delete  --nodes 192.168.64.25
```

**Pull request merged**